src/cloudwatch/modules/dimensionplugins/generic_dimensions.py

```diff
@@ -0,0 +1,53 @@
+"""
+This is the file containing generic dimension plugin classes
+"""
+
+import os
+from . import DimensionPlugin
+
+"""
+This InstanceId Dimension coming from configured host instance
+"""
+
+def dimension_get_instance_id(dimension, args):
+    dimension[args['name']] = args['value']
+
+class Dimension_InstanceId(DimensionPlugin):
+    def register_plugin(self):
+        self.func = dimension_get_instance_id
+        self.args = {
+            'name': "InstanceId",
+            'value': self.config.host
+        }
+
+
+"""
+This PluginInstance Dimension coming from collectd value plugin instance
+"""
+
+def dimension_get_plugin_instance(dimension, args):
+    dimension[args['name']] = args['value']
+
+class Dimension_PluginInstance(DimensionPlugin):
+    def register_plugin(self):
+        self.func = dimension_get_plugin_instance
+        self.args = {
+            'name': "PluginInstance",
+            'value': self.vl.plugin_instance
+        }
+
+
+"""
+Hostname Dimension report the hostname value
+"""
+
+def dimension_get_hostname(dimension, args):
+    dimension[args['name']] = args['value']
+
+class Dimension_Hostname(DimensionPlugin):
+    def register_plugin(self):
+        self.func = dimension_get_hostname
+        self.args = {
+            'name': "Hostname",
+            'value': os.uname()[1]
```

Comment:
Do not know if following issue will happen.
Return a 5-tuple containing information identifying the current operating system. The tuple contains 5 strings: (sysname, nodename, release, version, machine). Some systems truncate the nodename to 8 characters or to 
the leading component; a better way to get the hostname is socket.gethostname() or even socket.gethostbyaddr(socket.gethostname()).